wpimath/src/main/java/edu/wpi/first/math/optimization/SimulatedAnnealing.java

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.math.optimization;
+
+import edu.wpi.first.math.Num;
+import edu.wpi.first.math.Vector;
+import java.util.function.Function;
+
+/**
+ * An implementation of the Simulated Annealing stochastic nonlinear optimization method for finding
+ * the global minimum cost.
+ */
+public class SimulatedAnnealing {
+  /**
+   * Runs the Simulated Annealing algorithm on a Vector that defines the state.
+   *
+   * @param initialGuess The initial state
+   * @param epochs Number of iterations to run.
+   * @param initialTemperature The initial temperature
+   * @param getNeighbor Function to generate a random neighbor of current state vector.
+   * @param getCost Function to get the double value of a state vector.
```

Comment:
These need to be explained better.

What's an iteration in this context?
What's a temperature?
What's a neighbor in this context?

`getCost` seems like a cost function, but the description says otherwise because a cost function and a function that returns the state vector are not the same thing.